File: app/src/main/java/com/pietrantuono/podcasts/main/customviews/SimpleNavView.java
package com.pietrantuono.podcasts.main.customviews;

import android.content.Context;
import android.support.design.widget.NavigationView;
import android.util.AttributeSet;

import com.pietrantuono.podcasts.R;
import com.pietrantuono.podcasts.main.presenter.MainPresenter;

public class SimpleNavView extends NavigationView {
    private MainPresenter listener;
    private DrawerLayoutWithToggle drawerLayout;

    public SimpleNavView(Context context) {
        super(context);
        init();
    }

    public SimpleNavView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public SimpleNavView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        setNavigationItemSelectedListener(item -> {
            drawerLayout.closeDrawers();
            int id = item.getItemId();
            switch (id) {
                case R.id.add_podcast:
                    listener.onAddPodcastSelected();
                    break;
                case R.id.subscribed:
                    listener.onSubscribeSelected();
                    break;
            }
            return true;
        });
    }

    public void setMainPresenterListener(MainPresenter listener) {
        this.listener = listener;
    }

    public void setDrawer(DrawerLayoutWithToggle drawerLayout) {
        this.drawerLayout = drawerLayout;
    }
}

File: app/src/main/java/com/pietrantuono/podcasts/main/customviews/SimpleNavView.kt
package com.pietrantuono.podcasts.main.customviews

import android.content.Context
import android.support.design.widget.NavigationView
import android.util.AttributeSet

import com.pietrantuono.podcasts.R
import com.pietrantuono.podcasts.main.presenter.MainPresenter

class SimpleNavView : NavigationView {
    private var listener: MainPresenter? = null
    private var drawerLayout: DrawerLayoutWithToggle? = null

    //TODO use jvm overloads
    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init()
    }

    private fun init() {
        setNavigationItemSelectedListener { item ->
            drawerLayout?.closeDrawers()
            val id = item.itemId
            when (id) {
                R.id.add_podcast -> listener?.onAddPodcastSelected()
                R.id.subscribed -> listener?.onSubscribeSelected()
                R.id.settings -> listener?.onSettingsSelected()
                R.id.downloads -> listener?.onSettingsSelected()
            }
            true
        }
    }

    fun setMainPresenterListener(listener: MainPresenter) {
        this.listener = listener
    }

    fun setDrawer(drawerLayout: DrawerLayoutWithToggle) {
        this.drawerLayout = drawerLayout
    }
}
